tests/ui/inference/detect-old-time-version-format_description-parse.rs

```diff
@@ -0,0 +1,8 @@
+#![crate_name = "time"]
```

Comment:
Remark: *technically* I believe we would also trigger the diagnostic for any crates named `time` (not necessarily *the* `time` crate on crates.io), but as rustc for our intents and purposes we can't know in general, so I think it's fine.